ansible_base/rbac/caching.py

```diff
@@ -0,0 +1,190 @@
+import logging
+
+from ansible_base.rbac.models import ObjectRole, RoleDefinition, RoleEvaluation
+from ansible_base.rbac.permission_registry import permission_registry
+from ansible_base.rbac.prefetch import TypesPrefetch
+
+logger = logging.getLogger('ansible_base.rbac.caching')
+
+
+'''
+This module has callable methods to fill in things marked with COMPUTED DATA in the models
+from the user-specifications in other fields.
+These need to be called in specific hooks to assure that evaluations remain correct,
+logic for triggers are in the triggers module.
+
+NOTE:
+This is highly dependent on the model methods ObjectRole.needed_cache_updates and expected_direct_permissions
+Those methods are what truly dictate the object-role to object-permission translation
+'''
+
+
+def all_team_parents(team_id: int, team_team_parents: dict, seen=None) -> set:
+    """
+    Returns parent teams, and parent teams of parent teams, until we have them all
+        {parent_team_id, parent_team_id, ...}
+
+    team_id: id of the team we want to get the direct and indirect parents of
+    team_team_parents: mapping of team id to ids of its parents, this is not modified by this method
+    seen: mutable set that will be added to by each call so that we can not recurse infinitely
+    """
+    parent_team_ids = set()
+    if seen is None:
+        seen = set()
+    for parent_id in team_team_parents.get(team_id, []):
+        if parent_id in seen:
+            # will be combined in a lower level of the call stack
+            # this condition prevents infinite recursion in the event of loops in the graph
+            continue
+        parent_team_ids.add(parent_id)
+        seen.add(parent_id)
+        parent_team_ids.update(all_team_parents(parent_id, team_team_parents, seen=seen))
+    return parent_team_ids
+
+
+def get_org_team_mapping() -> dict:
```

Comment:
Would be nice to use more specific type hints (e.g. `dict[int, list[int]]` ) in the code.